### PR TITLE
Refactor of zipping functions

### PR DESCRIPTION
**Pull request by sfc-gh-jsikorski**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added new automated tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the section below.

### Changes description
Fix for #177 with refactoring of zip functions.
I separated tasks of creating a list of files to pack, and creating an archive, also moving duplicated code to separate function

### CODE REVIEW

File: src/snowcli/utils.py [hunk 1]
```diff
@@ -478,66 +477,59 @@ def install_packages(
...
-                    ):
-                        zipf.write(file, arcname=os.path.relpath(file, directory))
-
-    # close the zip file object
-    zipf.close()
+    files_to_pack = get_list_of_files_to_pack(pack_dir, True)
+
+    with ZipFile(dest_zip, "w", ZIP_DEFLATED, allowZip64=True) as package_zip:
+        for file in files_to_pack:
+            package_zip.write(file[0], arcname=os.path.relpath(file[0], file[1]))
```
Comment (sfc-gh-turbaszek) [inline src/snowcli/utils.py:484]: Do we need to use `file[0]` ? Can't we use `file.name` or something to be more explicite?

File: src/snowcli/utils.py [hunk 2]
```diff
@@ -478,66 +477,59 @@ def install_packages(
...
+
+    files: List[Tuple[Path, Optional[str]]] = []
+
+    def filenames_filter(file: Path) -> bool:
+        return (
+            not file.name.startswith(".")
+            and not str(file).startswith(".")
+            and not file.match("*.pyc")
+            and not file.match("*__pycache__*")
+            and file not in map(lambda x: x[0], files)
```
Comment (sfc-gh-turbaszek) [inline src/snowcli/utils.py:509]:
```suggestion
            and file not in map(lambda x: x[0], excluded_files)
```
WDYT? Also, `[f[0] for f in files]` would be probably more pythonic

File: src/snowcli/utils.py (hunk 1 above)
Comment (sfc-gh-jsikorski) [reply src/snowcli/utils.py]: I added File dataclass - now it's file.name

File: src/snowcli/utils.py (hunk 2 above)
Comment (sfc-gh-turbaszek) [reply src/snowcli/utils.py]: MAybe we can use `set` instead of list? It automatically resolves the duplicates. WDYT?

Comment (sfc-gh-jsikorski) [reply src/snowcli/utils.py]: I thought about it, but, when added in different steps, Files can have the same name and different relpath - therefore a simple change from list to set would potentially leave some duplicates.
And, as we`re already checking the filename, i thought adding the check there would the simplest solution.

File: src/snowcli/utils.py [hunk 3]
```diff
@@ -478,66 +477,67 @@ def install_packages(
...
-
-    # close the zip file object
-    zipf.close()
+    files_to_pack = get_list_of_files_to_pack(pack_dir, True)
+
+    with ZipFile(dest_zip, "w", ZIP_DEFLATED, allowZip64=True) as package_zip:
+        for file in files_to_pack:
+            package_zip.write(
+                file.name, arcname=os.path.relpath(file.name, file.relpath)
+            )
```
Comment (sfc-gh-astus) [inline src/snowcli/utils.py:486]: Great candidate to extract it to separate function. In `standard_zip_dir` you have the same code

Comment (sfc-gh-jsikorski) [reply src/snowcli/utils.py]: Good catch - created separate function

**Pull request merged**